product_pricelist_supplierinfo/models/product_product.py

```diff
@@ -45,8 +45,8 @@ def _select_seller(
                 lambda s: not s.company_id or s.company_id.id == company_id)
         for seller in sellers:
             if (
-                (not seller.date_start or seller.date_start <= date)
-                and (not seller.date_end or seller.date_end >= date)
+                (not seller.date_start or fields.context_today(seller, seller.date_start) <= date)
```

Comment:
```suggestion
                (not seller.date_start or fields.Date.context_today(seller, datetime.combine(seller.date_start, datetime.min.time()) + timedelta(hours=6)) <= date)
```

Vas a necesitar importar al incio del archivo

`from datetime import datetime, timedelta`